File: ChatApp-Python/load_test_client.py
import socket
import pickle
import threading
import time
from chat_message import ChatMessage

def simulate_client(client_id, messages_per_client, delay_between_messages):
    try:
        # Conectar al servidor
        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        client_socket.connect(("localhost", 1500))

        # Enviar nombre de usuario
        username = f"User{client_id}"
        client_socket.sendall(pickle.dumps(username))

        # Enviar mensajes
        for i in range(messages_per_client):
            message = f"Message {i} from {username}"
            start_time = time.time()  # Tiempo antes de enviar el mensaje
            client_socket.sendall(pickle.dumps(ChatMessage(ChatMessage.MESSAGE, message)))
            end_time = time.time()  # Tiempo después de enviar el mensaje
            with open("resultados_python.txt", "a") as file:
                file.write(f"{(end_time - start_time) * 1000:.2f}\n")
            time.sleep(delay_between_messages / 1000.0)

        # Desconectar
        client_socket.sendall(pickle.dumps(ChatMessage(ChatMessage.LOGOUT, "")))
        client_socket.close()
    except Exception as e:
        print(f"Error en cliente {client_id}: {e}")

    finally:
        if client_socket:
            client_socket.close()

def main():
    messages_per_client = 10  # Número de mensajes por cliente
    delay_between_messages = 100  # Milisegundos entre mensajes

    for num_clients in range(100, 1001, 100):
        print(f"Probando con {num_clients} clientes...")
        threads = []
        for i in range(num_clients):
            t = threading.Thread(target=simulate_client, args=(i, messages_per_client, delay_between_messages))
            threads.append(t)
            t.start()

        for t in threads:
            t.join()

if __name__ == "__main__":
    main()

File: ChatApp/src/LoadTestClient.java
import java.io.*;
import java.net.*;
import java.util.concurrent.*;

public class LoadTestClient {
    public static void main(String[] args) {
        int messagesPerClient = 10; // Número de mensajes por cliente
        int delayBetweenMessages = 100; // Milisegundos entre mensajes

        for (int numClients = 100; numClients <= 1000; numClients += 100) {
            System.out.println("Probando con " + numClients + " clientes...");
            ExecutorService executor = Executors.newFixedThreadPool(numClients);

            for (int i = 0; i < numClients; i++) {
                int clientId = i;
                executor.execute(() -> {
                    Socket socket = null;
                    ObjectOutputStream sOutput = null;
                    ObjectInputStream sInput = null;
                    try {
                        socket = new Socket("localhost", 1500);
                        sOutput = new ObjectOutputStream(socket.getOutputStream());
                        sInput = new ObjectInputStream(socket.getInputStream());

                        // Enviar nombre de usuario
                        String username = "User" + clientId;
                        sOutput.writeObject(username);

                        // Enviar mensajes
                        for (int j = 0; j < messagesPerClient; j++) {
                            String message = "Message " + j + " from " + username;
                            long startTime = System.currentTimeMillis(); // Tiempo antes de enviar el mensaje
                            sOutput.writeObject(new ChatMessage(ChatMessage.MESSAGE, message));
                            long endTime = System.currentTimeMillis(); // Tiempo después de enviar el mensaje
                            try (PrintWriter writer = new PrintWriter(new FileWriter("resultados_java.txt", true))) {
                                writer.println(endTime - startTime);
                            } catch (IOException e) {
                                e.printStackTrace();
                            }
                            Thread.sleep(delayBetweenMessages);
                        }

                        // Desconectar
                        sOutput.writeObject(new ChatMessage(ChatMessage.LOGOUT, ""));
                    } catch (Exception e) {
                        System.out.println("Error en cliente " + clientId + ": " + e.getMessage());
                    } finally {
                        try {
                            if (sOutput != null) sOutput.close();
                            if (sInput != null) sInput.close();
                            if (socket != null) socket.close();
                        } catch (IOException e) {
                            System.out.println("Error cerrando recursos para cliente " + clientId + ": " + e.getMessage());
                        }
                    }
                });
            }

            executor.shutdown();
            try {
                executor.awaitTermination(1, TimeUnit.MINUTES);
            } catch (InterruptedException e) {
                System.out.println("Error esperando la finalización de los clientes.");
            }
        }
    }
}